### PR TITLE
Add support for Razer Basilisk Essential

### LINKED ISSUE #1134: Support for Razer Basilisk Essential
saw support added for the Razer Basilisk in #804fa58 yet still no support for the Essential model 
### Device Info

* Name: Razer Basilisk Essential 
* Product Number: RZ01-02650100
* Model Number: RZ01-0265

### Device Information

output of lsusb -d 1532:0065 -v

```
Bus 002 Device 004: ID 1532:0065 Razer USA, Ltd 
Device Descriptor:
  bLength                18
  bDescriptorType         1
  bcdUSB               2.00
  bDeviceClass            0 
  bDeviceSubClass         0 
  bDeviceProtocol         0 
  bMaxPacketSize0        64
  idVendor           0x1532 Razer USA, Ltd
  idProduct          0x0065 
  bcdDevice            2.00
  iManufacturer           1 Razer
  iProduct                2 Razer Basilisk Essential
  iSerial                 0 
  bNumConfigurations      1
  Configuration Descriptor:
    bLength                 9
    bDescriptorType         2
    wTotalLength       0x0054
    bNumInterfaces          3
    bConfigurationValue     1
    iConfiguration          0 
    bmAttributes         0xa0
      (Bus Powered)
      Remote Wakeup
    MaxPower              500mA
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        0
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      1 Boot Interface Subclass
      bInterfaceProtocol      2 Mouse
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.11
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength      94
         Report Descriptors: 
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x81  EP 1 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0008  1x 8 bytes
        bInterval               1
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        1
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      0 
      bInterfaceProtocol      1 Keyboard
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.11
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength     159
         Report Descriptors: 
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x82  EP 2 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0010  1x 16 bytes
        bInterval               1
    Interface Descriptor:
      bLength                 9
      bDescriptorType         4
      bInterfaceNumber        2
      bAlternateSetting       0
      bNumEndpoints           1
      bInterfaceClass         3 Human Interface Device
      bInterfaceSubClass      0 
      bInterfaceProtocol      1 Keyboard
      iInterface              0 
        HID Device Descriptor:
          bLength                 9
          bDescriptorType        33
          bcdHID               1.11
          bCountryCode            0 Not supported
          bNumDescriptors         1
          bDescriptorType        34 Report
          wDescriptorLength      61
         Report Descriptors: 
           ** UNAVAILABLE **
      Endpoint Descriptor:
        bLength                 7
        bDescriptorType         5
        bEndpointAddress     0x83  EP 3 IN
        bmAttributes            3
          Transfer Type            Interrupt
          Synch Type               None
          Usage Type               Data
        wMaxPacketSize     0x0008  1x 8 bytes
        bInterval               1
can't get device qualifier: Resource temporarily unavailable
can't get debug descriptor: Resource temporarily unavailable
Device Status:     0x0000
  (Bus Powered)

```

```
lrwxrwxrwx 1 root root  10 May  2 22:20 usb-Razer_Razer_Basilisk_Essential-event-if01 -> ../event12
lrwxrwxrwx 1 root root   9 May  2 22:20 usb-Razer_Razer_Basilisk_Essential-event-mouse -> ../event8
lrwxrwxrwx 1 root root   9 May  2 22:20 usb-Razer_Razer_Basilisk_Essential-if01-event-kbd -> ../event9
lrwxrwxrwx 1 root root  10 May  2 22:20 usb-Razer_Razer_Basilisk_Essential-if02-event-kbd -> ../event13
lrwxrwxrwx 1 root root   9 May  2 22:20 usb-Razer_Razer_Basilisk_Essential-mouse -> ../mouse0

```

### Packet Captures
currently not possible 
### Synapse Screenshots
image
image
image
image
image

### PR DESCRIPTION
**Pull request by slugcat-dev**
**Fixes #1134**

I've already tested everything on my PC, an additional review would be useful.

**Mouse settings:**
- DPI (Y-axis doesn't affect anything, but the option cannot be removed)
- Polling rate

**Logo effects:**
- Breath
- Spectrum
- Reactive
- Static
- None

### CODE REVIEW

**Comment by slugcat-dev** [comment]
Ready to merge

**Comment by terricain** [comment]
@z3ntu this looks ok to me, thoughts?

File: daemon/openrazer_daemon/hardware/mouse.py [hunk 1]
```diff
@@ -2071,6 +2071,54 @@ def _resume_device(self):
...
+        _da_set_logo_brightness(self, 0)
+        self.disable_notify = False
+
+    def _resume_device(self):
+        """
+        Resume the device
+
+        Get the last known brightness and then set the brightness
+        """
+        logo_brightness = self.suspend_args.get('brightness', (100, 100))[0]
```
Comment (z3ntu) [inline daemon/openrazer_daemon/hardware/mouse.py]: This should be `logo_brightness = self.suspend_args.get('brightness', 100)` as you're only putting one value into the variable in `_suspend_device()`

**Review by z3ntu — CHANGES_REQUESTED** [review]
dpi_stages is declared as supported in the daemon but not present in the driver. The device should support that feature so please eloborate.

**Review by z3ntu — APPROVED** [review]
Diff looks good. Just to confirm again, all the functionality is working?

**Comment by slugcat-dev** [comment]
Yep, just tested it again, works fine

File: driver/razermouse_driver.c [hunk 2]
```diff
@@ -4777,6 +4812,18 @@ static void razer_mouse_disconnect(struct hid_device *hdev)
...
+            device_remove_file(&hdev->dev, &dev_attr_poll_rate);
+            device_remove_file(&hdev->dev, &dev_attr_dpi);
+            device_remove_file(&hdev->dev, &dev_attr_dpi_stages);
+            device_remove_file(&hdev->dev, &dev_attr_logo_led_brightness);
+            device_remove_file(&hdev->dev, &dev_attr_logo_matrix_effect_spectrum);
+            device_remove_file(&hdev->dev, &dev_attr_logo_matrix_effect_reactive);
+            device_remove_file(&hdev->dev, &dev_attr_logo_matrix_effect_breath);
+            device_remove_file(&hdev->dev, &dev_attr_logo_matrix_effect_static);
+            device_remove_file(&hdev->dev, &dev_attr_logo_matrix_effect_none);
+            break;
```
Comment (z3ntu) [inline driver/razermouse_driver.c]: you forgot these lines here but I'll fix it for you
```c
            device_remove_file(&hdev->dev, &dev_attr_matrix_effect_custom);
            device_remove_file(&hdev->dev, &dev_attr_matrix_custom_frame);
```

Comment (slugcat-dev) [reply driver/razermouse_driver.c]: Thank you

**Review by z3ntu — APPROVED** [review]
LGTM, thanks!

**Pull request merged**